Makefile: generate coverage for privileged unit tests

christarazi opened the pull request:
See commit messages.

christarazi committed:
Makefile: generate coverage for privileged tests

The `tests-privileged` target previously did not generate coverage.
This commit adds the ability to generate coverage which is useful when
developing on privileged tests.

Signed-off-by: Chris Tarazi <chris@isovalent.com>

christarazi committed:
Makefile: new target to share coverage init logic

This removes the duplication between normal unprivileged unit tests and
privileged unit tests.

Signed-off-by: Chris Tarazi <chris@isovalent.com>

joestringer commented on Makefile:
```diff
@@ -121,8 +121,16 @@ PRIV_TEST_PKGS ?= $(PRIV_TEST_PKGS_EVAL)
+		tail -n +2 coverage.out >> coverage-all-tmp.out; \
+	done
+	$(MAKE) generate-cov
+	@rmdir ./daemon/1 ./daemon/1_backup 2> /dev/null || true
```
Did you find that this step was necessary here as well or is it just following the existing code?

Also wondering (1) why this even exists; and if it needs to, (2) why not put it as part of the `generate-cov` target.

christarazi replied:
I was following existing code. I was trying to do some git archaeology to understand why it was there to begin with, but I was only able to get to this commit (33244acae) which doesn't explain why it's needed. I presumed some tests left state and needed to be cleaned up, and I didn't go further than that.

Anyway, moving it to the `generate-cov` target is a good idea.

joestringer reviewed (approved):
Looks good. One minor nit that could be cleaned up below.

christarazi committed:
Makefile: move dir cleanup to generate-cov target

This simplifies the unit-tests and test-privileged as they both had this
logic after generating coverage.

Signed-off-by: Chris Tarazi <chris@isovalent.com>
